Shell/LineEditor.cpp

```diff
@@ -360,9 +374,26 @@ String LineEditor::get_line(const String& prompt)
                 else
                     tab_complete_other_token(token);
 
+                if (m_times_tab_pressed > 1 && !m_suggestions.is_empty()) {
+                    for (auto i = 0; i < m_suggestions.size(); i++) {
+                        if (i % 3 == 0)
+                            putchar('\n');
+
+                        printf("%-20s", m_suggestions.at(i).characters());
+                    }
+
+                    putchar('\n');
+                    fputs(prompt.characters(), stdout);
+                    fputs(String(m_buffer.data(), m_cursor).characters(), stdout);
```

Comment:
No need for a temporary string here, we can just `write()`:
```cpp
write(STDOUT_FILENO, m_buffer.data(), m_cursor);
```

Reply:
Well technically, no, because `write()` can be interrupted or just write out less than requested, and stdio is expected to deal with that IIRC.